src/web/crate_details.rs

```diff
@@ -466,6 +470,160 @@ pub(crate) async fn get_all_releases(
+            if let Some(name) = release_found.corrected_name {
+                return redirect(&name, "latest", &req_path, CachePolicy::NoCaching);
+            }
+
+            (version, "latest".to_string())
+        }
+
+        // Redirect when the requested version isn't correct
+        MatchSemver::Semver((v, _)) => {
+            // to prevent cloudfront caching the wrong artifacts on URLs with loose semver
+            // versions, redirect the browser to the returned version instead of loading it
+            // immediately
+            return redirect(&params.name, &v, &req_path, CachePolicy::ForeverInCdn);
+        }
+    };
+
+    let (name, doc_targets, releases, default_target): (String, Vec<String>, Vec<Release>, String) =
+        spawn_blocking({
+            let pool = pool.clone();
+            move || {
+                let mut conn = pool.get()?;
+                let query = "
+            SELECT
+                crates.id,
+                crates.name,
+                releases.default_target,
+                releases.doc_targets
+            FROM releases
+            INNER JOIN crates ON releases.crate_id = crates.id
+            WHERE crates.name = $1 AND releases.version = $2;";
+
+                let rows = conn.query(query, &[&params.name, &version])?;
+
+                let krate = if rows.is_empty() {
+                    return Err(AxumNope::CrateNotFound.into());
+                } else {
+                    &rows[0]
+                };
+
+                // get releases, sorted by semver
+                let releases = releases_for_crate(&mut *conn, krate.get("id"))?;
+
+                Ok((
+                    krate.get("name"),
+                    MetaData::parse_doc_targets(krate.get("doc_targets")),
+                    releases,
+                    krate.get("default_target"),
+                ))
+            }
+        })
+        .await?;
+
+    let latest_release = releases
+        .iter()
+        .find(|release| release.version.pre.is_empty() && !release.yanked)
+        .unwrap_or(&releases[0]);
+
+    // The path within this crate version's rustdoc output
+    let (target, inner_path) = {
+        let mut inner_path = req_path.clone();
+
+        let target = if inner_path.len() > 1 && doc_targets.iter().any(|s| s == inner_path[0]) {
+            inner_path.remove(0)
+        } else {
+            ""
+        };
+
+        (target, inner_path.join("/"))
+    };
+
+    let current_target = if latest_release.build_status {
+        if target.is_empty() {
+            default_target
+        } else {
+            target.to_owned()
+        }
+    } else {
+        String::new()
+    };
+
+    let res = PlatformList {
+        metadata: ShortMetadata {
+            name,
+            version_or_latest: version_or_latest.to_string(),
+            doc_targets,
+        },
+        inner_path,
+        use_direct_platform_links: true,
```

Comment:
This shouldn't be unconditionally true, it should only be set when we're linking to the crate root (I think that's `inner_path == format!("{crate_name}/index.html")` by the looks of it, we normalize the path enough to not have to check multiple variations).

Reply:
So anything to change here?